### PR TITLE
Add spec.secretKey for access to raw broker response

### PR DESCRIPTION
**Pull request by phil9909**
With this change, setting the optional `spec.secretKey` will cause the reconciler to store the raw broker response json in the given key (inside the secret).
If it is not set, the old behavior will be used (parse the response and use the top level keys in `secret.Data`).

Motivation can be found in Issue #61

Alternative: We could put `spec.secretName` and `spec.secretKey` in one object and use `spec.secret.name` and `spec.secret.key`, but this would break the API.

### CODE REVIEW

File: controllers/servicebinding_controller_test.go [hunk 1]
```diff
@@ -818,7 +851,7 @@ var _ = Describe("ServiceBinding controller", func() {
 
 func validateSecretData(secret *v1.Secret, expectedKey string, expectedValue string) {
 	Expect(secret.Data).ToNot(BeNil())
-	Expect(len(secret.Data[expectedKey])).ToNot(BeNil())
+	Expect(secret.Data).To(HaveKey(expectedKey))
```
Comment (phil9909) [inline controllers/servicebinding_controller_test.go:854]: This has nothing to do with the actual PR. But it seemed wrong to us:
`len(...)` will always return an int, therefore the `Expect` would always succeed.

File: api/v1alpha1/servicebinding_types.go [hunk 2]
```diff
@@ -44,6 +44,13 @@ type ServiceBindingSpec struct {
 	// +optional
 	SecretName string `json:"secretName"`
 
+	// SecretKey is used as the key inside the secret to store the credentials
+	// returned by the broker encoded as json. If not specified the credentials
+	// returned by the broker will be used directly as the secrets data, which
+	// can cause problems when using complex data structures.
```
Comment (eyalnathan) [inline api/v1alpha1/servicebinding_types.go:50]: can you replace to a more positive phrasing? e.g. "SecretKey is used as the key inside the secret to store the credentials returned by the broker encoded as json to support complex data structures. If not specified the credentials returned by the broker will be used directly as the secrets data."

**Review by eyalnathan — CHANGES_REQUESTED** [review]
Code looks good, please see my comment regarding the docu of the new property

File: api/v1alpha1/servicebinding_types.go (hunk 2 above)
Comment (phil9909) [reply api/v1alpha1/servicebinding_types.go]: I used your suggestion.

**Review by eyalnathan — APPROVED** [review]

**Comment by pavelmaliy** [comment]
@phil9909 you can upgrade to v0.1.10 version with the latest commit

**Pull request merged**